src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -121,7 +121,8 @@ public Conversion ClassifyImplicitConversionFromExpression(BoundExpression sourc
             }
             else if (sourceExpression.GetFunctionType() is { } sourceFunctionType)
             {
-                if (HasImplicitFunctionTypeConversion(sourceFunctionType, destination, ref useSiteInfo))
+                if (HasImplicitFunctionTypeConversion(sourceFunctionType, destination, ref useSiteInfo) &&
+                    sourceFunctionType.GetInternalDelegateType() is { })
```

Comment:
>sourceFunctionType.GetInternalDelegateType() is { } [](http://example.com/codeflow?start=20&length=51)

I think it would be better if ```HasImplicitFunctionTypeConversion``` checked for that when appropriate.